mlir/lib/Dialect/Affine/Utils/LoopFusionUtils.cpp

```diff
@@ -275,7 +293,13 @@ FusionResult mlir::affine::canFuseLoops(AffineForOp srcForOp,
   // 'forOpA' executes before 'forOpB' in 'block'.
   auto forOpA = isSrcForOpBeforeDstForOp ? srcForOp : dstForOp;
   auto forOpB = isSrcForOpBeforeDstForOp ? dstForOp : srcForOp;
-
+  // Check if either of the loop has any possible side-effects.
+  // if yes then fusion precondition failed.
+  if (hasPossibleSideEffects(forOpA) || hasPossibleSideEffects(forOpB)) {
```

Comment:
Actually, containing non-affine side-effecting op by itself doesn't make the fusion invalid. In some cases, it's still valid. It's when the fusion leads to redundant computation. So you need to mention this is conservative if you are simply disabling it based on the existence of such a side-effecting op.